### ISSUE TITLE
implicitly-declared ‘XrdXrootdProtocol::XrdXrootdProtocol(XrdXrootdProtocol&)’ is deprecated

### ISSUE THREAD

**Issue by vss888** [xrootd/xrootd#1000]
I get the following error while trying to compile ROOT:

```
[I] No git repository info found. Trying to interpret VERSION_INFO
[I] src/XrdVersion.hh successfully generated
.../root-6.16.00.build/XROOTD-prefix/src/XROOTD/src/XrdXrootd/XrdXrootdProtocol.cc: In member function ‘XrdXrootdProtocol XrdXrootdProtocol::operator=(const XrdXrootdProtocol&)’:
.../root-6.16.00.build/XROOTD-prefix/src/XROOTD/src/XrdXrootd/XrdXrootdProtocol.cc:234:12: error: implicitly-declared ‘XrdXrootdProtocol::XrdXrootdProtocol(XrdXrootdProtocol&)’ is deprecated [-Werror=deprecated-copy]
  234 |    return *this;
      |            ^~~~
.../root-6.16.00.build/XROOTD-prefix/src/XROOTD/src/XrdXrootd/XrdXrootdProtocol.cc:215:19: note: because ‘XrdXrootdProtocol’ has user-provided ‘XrdXrootdProtocol XrdXrootdProtocol::operator=(const XrdXrootdProtocol&)’
  215 | XrdXrootdProtocol XrdXrootdProtocol::operator =(const XrdXrootdProtocol &rhs)
      |                   ^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[5]: *** [src/CMakeFiles/XrdServer.dir/XrdXrootd/XrdXrootdProtocol.cc.o] Error 1
gmake[4]: *** [src/CMakeFiles/XrdServer.dir/all] Error 2
gmake[3]: *** [all] Error 2
```
Could you please let me know if there is anything that can be done? Using gcc-9.1.0 to compile it. Has anybody tried to compile Xrootd with this version of gcc?

Thank you for your help!

**Comment by simonmichal** [comment]
@vss888 : this has been fixed in 7c5fc93b1bc80b98cb5b13faa82908c426bc4e0c, for the record we do have builds for fedora-rawhide (which has the latest gcc) in our CI, e.g.: https://gitlab.cern.ch/dss/xrootd/-/jobs/4503990

If this answers your question, please close this issue.

**Comment by vss888** [comment]
@simonmichal Thank you for your reply! Is there an easy way to compile ROOT with the master version of Xrootd?

**Comment by simonmichal** [comment]
@vss888 : if you are building xrootd together with root you could change the cmake so it takes the latest xrootd tag instead of a tar ball source, something like:

```
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -769,8 +769,10 @@ if(builtin_xrootd)
                        ${XROOTD_ROOTDIR}/${_LIBDIR_DEFAULT}/libXrdCl${CMAKE_SHARED_LIBRARY_SUFFIX})
   ExternalProject_Add(
     XROOTD
-    URL ${XROOTD_SRC_URI}
-    URL_HASH SHA256=42e4d2cc6f8b442135f09bcc12c7be38b1a0c623a005cb5e69ff3d27997bdf73
+#    URL ${XROOTD_SRC_URI}
+#    URL_HASH SHA256=42e4d2cc6f8b442135f09bcc12c7be38b1a0c623a005cb5e69ff3d27997bdf73
+    GIT_REPOSITORY  https://github.com/xrootd/xrootd.git
+    GIT_TAG         v4.10.0-rc2
     INSTALL_DIR ${XROOTD_ROOTDIR}
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
                -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
@@ -781,6 +783,7 @@ if(builtin_xrootd)
                -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
                -DENABLE_PYTHON=OFF
                -DENABLE_CEPH=OFF
+               -DXRDCL_ONLY=ON
     INSTALL_COMMAND ${CMAKE_COMMAND} --build . --target install
             COMMAND ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/include/xrootd <INSTALL_DIR>/include
     LOG_DOWNLOAD 1 LOG_CONFIGURE 1 LOG_BUILD 1 LOG_INSTALL 1

```

**Comment by vss888** [comment]
@simonmichal Thank you very much for the advise! I took a little bit easier route inspired by it:

1. Replaced `$ROOTSOURCE/cmake/modules/SearchInstalledSoftware.cmake` around line 882
   `set(XROOTD_VERSION 4.8.5)` with `set(XROOTD_VERSION 4.9.1)`
2. Replaced 
`URL_HASH SHA256=42e4d2cc6f8b442135f09bcc12c7be38b1a0c623a005cb5e69ff3d27997bdf73`
with
`URL_HASH SHA256=de47160348d52a655f8d601520ad9146b9ee76eb5d2352cdf232a6fafce27adb`

and the compilation of Xrootd seems to have succeeded.
